packages/@ember/-internals/utils/lib/cache.ts

```diff
@@ -3,15 +3,17 @@ export default class Cache<T, V> {
   public misses = 0;
   public hits = 0;
 
-  constructor(private limit: number, private func: (obj: T) => V, private store?: any) {
-    this.store = store || new Map();
-  }
+  constructor(
+    private limit: number,
+    private func: (obj: T) => V,
+    private store: Map<T, V> = new Map()
+  ) {}
 
   get(key: T): V {
-    if (this.store.has(key)) {
+    let value = this.store.get(key);
+    if (value) {
```

Comment:
This assumes we never cache a falsy value. I assume you've already verified that.

Reply:
Ah, that's interesting. 🤔 I'll double check—and that may actually be the cause of the failure, and the reason we were "double"-checking before.

Reply:
That was indeed the thing. Resolved by reverting that behavior. «sigh»